src/provider/rpc.ts

```diff
@@ -59,9 +73,18 @@ export class RpcProvider implements ProviderInterface {
 
   private chainId?: StarknetChainId;
 
-  constructor(optionsOrProvider: RpcProviderOptions) {
-    const { nodeUrl, retries, headers, blockIdentifier, chainId } = optionsOrProvider;
-    this.nodeUrl = nodeUrl;
+  constructor(optionsOrProvider?: RpcProviderOptions) {
+    const { nodeUrl, retries, headers, blockIdentifier, chainId } = optionsOrProvider || {};
+    if ((<any>Object).values(NetworkName).includes(nodeUrl)) {
```

Comment:
Could also do:
```suggestion
    if (Object.values(NetworkName as any).includes(nodeUrl)) {
```
or 
```suggestion
    if (Object.values(NetworkName).includes(nodeUrl as NetworkName)) {
```